tools/image_processing/imagej2/readme.md

```diff
@@ -0,0 +1,102 @@
+Galaxy wrappers for ImaageJ2 tools
+==================================
+
+ImageJ2 is a new version of ImageJ for the next generation of multidimensional image data, with a focus on scientific imaging. Its central goal is to broaden the paradigm of ImageJ beyond the limitations of ImageJ 1.x, to support the next generation of multidimensional scientific imaging.
+
+Fiji is an image processing package. It can be described as a "batteries-included" distribution of ImageJ (and ImageJ2), bundling Java, Java3D and a lot of plugins organized into a coherent menu structure. Fiji compares to ImageJ as Ubuntu compares to Linux.
+
+More informations is available at:
+
+* [http://fiji.sc/ImageJ2](http://fiji.sc/ImageJ2)
+* [http://fiji.sc/Fiji](http://fiji.sc/Fiji)
+
+
+Installation
+============
+
+Galaxy tool wrappers use specified Fiji Lifeline versions available from [http://fiji.sc/Downloads](http://fiji.sc/Downloads).  Galaxy should be able to automatically install this package.
+
+The wrappers are available at [https://github.com/bgruening/galaxytools/tree/master/tools/image_processing/imagej2](https://github.com/bgruening/galaxytools/tree/master/tools/image_processing/imagej2).
+
+
+Use Docker
+==========
+
+A docker image that installs Galaxy with these imaging tools is available at [https://github.com/bgruening/galaxy-imaging](https://github.com/bgruening/galaxy-imaging).
+
+
+Using Fiji with Galaxy tools
+============================
+
+Galaxy ImageJ2 tool wrappers generate a command line that calls a Python script, passing it a series of arguments including a Jython script named jython_script.py that resides in the same directory as the tool wrapper.  During tool execution, the Python script will call ImageJ2 with the --headless argument to run without the ImageJ2 GUI.  The Jython script is also passed to ImageJ2 along with all command line arguments that it expects.  ImageJ2 will execute the Jython script, passing the expected arguments.  The command line to run ImageJ2 from a Galaxy tool wrapper looks something like this:
+
+`ImageJ2 --ij2 --headless --jython ~jython_script.py arg1, arg2, ...`
+
+Each tool execution starts the ImageJ2 application within a Java virtual machine (JVM).  When ImageJ2 is finished processing the Jython script, the results are either written to a file or returned to the calling Galaxy process.  The JVM is shut down, and the Galaxy job terminates.
```

Comment:
I would squeeze HPC somewhere in here. With this in place you can run your tools on any supported Galaxy HPC env.
Also it's probably nice to point out, that we can only support `--headless` IJ plugins.